chia/consensus/block_header_validation.py

```diff
@@ -858,11 +858,11 @@ def validate_finished_header_block(
         blocks,
         unfinished_header_block,
         check_filter,
-        expected_difficulty,
-        expected_sub_slot_iters,
+        vs.current_difficulty,
+        vs.current_ssi,
         False,
         check_sub_epoch_summary=check_sub_epoch_summary,
-        prev_ses_block=prev_ses_block,
+        prev_ses_block=vs.prev_ses_block,
```

Comment:
It looks like `validate_unfinished_header_block()` also should just take `ValidationState`

Reply:
I have that one in a separate branch as I don't want to alter it in this one.